### PR TITLE
Created repository class and implemented fetchTrendingMovies function

### PR DESCRIPTION
**Pull request by Obehi234**
Repository set up ongoing

### CODE REVIEW

File: app/src/main/java/com/example/streammoviesapplication/data/repository/TrendingMoviesRepository.kt
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/repository/TrendingMoviesRepository.kt:1]: Lets create an interface for the repository, add the functions, then override it in this class. Research on how its done

File: app/src/main/java/com/example/streammoviesapplication/data/repository/TrendingMoviesImpl.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.streammoviesapplication.data.repository
+
+import com.example.streammoviesapplication.data.trendingMovies.localData.TrendingMoviesEntity
+import kotlinx.coroutines.flow.Flow
+
+interface TrendingMoviesImpl{
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/repository/TrendingMoviesImpl.kt:6]: rename to TrendingMoviesRepository

File: app/src/main/java/com/example/streammoviesapplication/data/repository/TrendingMoviesRepository.kt [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
 import com.example.streammoviesapplication.data.db.TrendingMoviesDao
+import com.example.streammoviesapplication.data.trendingMovies.localData.TrendingMoviesEntity
 import com.example.streammoviesapplication.network.MovieService
+import com.example.streammoviesapplication.utils.MovieMapper
+import kotlinx.coroutines.flow.Flow
 import javax.inject.Inject
 
-class TrendingMoviesRepository @Inject constructor(private val api: MovieService,
-private val trendingMoviesDao: TrendingMoviesDao) {
+class TrendingMoviesRepository
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/repository/TrendingMoviesRepository.kt:12]: rename to TrendingMoviesRepositoryImpl

File: app/src/main/java/com/example/streammoviesapplication/data/db/TrendingMoviesDao.kt [hunk 3]
```diff
@@ -14,7 +15,7 @@ interface TrendingMoviesDao {
     suspend fun insertTrendingMovies(movies: List<TrendingMoviesEntity>)
```
Comment (Ruthy4) [inline app/src/main/java/com/example/streammoviesapplication/data/db/TrendingMoviesDao.kt:15]: you didnt add  @Insert(onConflict = OnConflictStrategy.REPLACE)

Comment (Obehi234) [reply app/src/main/java/com/example/streammoviesapplication/data/db/TrendingMoviesDao.kt:15]: i did o

**Review by Ruthy4 — APPROVED** [review]

**Pull request merged**